src/main/java/io/rpg/wrapper/WrapperController.java

```diff
@@ -0,0 +1,150 @@
+package io.rpg.wrapper;
+
+import com.kkafara.rt.Result;
+import io.rpg.Game;
+import io.rpg.Initializer;
+import io.rpg.Main;
+import java.io.IOException;
+import java.util.Optional;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextArea;
+import javafx.scene.layout.BorderPane;
+import javafx.stage.Stage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Adds start and load screens.
+ */
+public class WrapperController {
+  private final Logger logger = LogManager.getLogger(WrapperController.class);
+  @FXML private BorderPane mainPane;
+  @FXML private TextArea outputArea;
+  @FXML private Button startButton;
+  private Stage stage;
+  private Scene scene;
+  private Parent startView;
+  private Parent loadView;
+
+  private final ConfigChooser chooser;
+  private Game game;
+
+  public WrapperController() {
+    chooser = new ConfigChooser();
+  }
+
+  public static WrapperController load() throws IOException {
+    FXMLLoader loader = new FXMLLoader(WrapperController.class.getResource("main-view.fxml"));
+    loader.load();
+    WrapperController controller = loader.getController();
+
+    controller.loadStartView();
+    controller.loadLoadView();
+
+    controller.createScene();
+    return controller;
+  }
+
+  private void loadStartView() throws IOException {
+    FXMLLoader loader = new FXMLLoader(WrapperController.class.getResource("start-view.fxml"));
+    loader.setController(this);
+    this.startView = loader.load();
+  }
+
+  private void loadLoadView() throws IOException {
+    FXMLLoader loader = new FXMLLoader(WrapperController.class.getResource("load-view.fxml"));
+    loader.setController(this);
+    this.loadView = loader.load();
+  }
+
+  private void createScene() {
+    scene = new Scene(mainPane);
+  }
+
+  public void show(Stage stage) {
+    this.stage = stage;
+    showView(startView);
+    stage.setScene(scene);
+    stage.show();
+  }
+
+  private void showView(Parent view) {
+    mainPane.setCenter(view);
+  }
+
+  @FXML
+  private void onBeginClick() {
+    showView(loadView);
+  }
+
+  @FXML
+  private void onExitClick() {
+    stage.close();
+  }
+
+  @FXML
+  private void onBackClick() {
+    showView(startView);
+  }
+
+  @FXML
+  private void onLoadClick() {
+    Optional<String> pathOptional = this.chooser.open(stage);
+    if (pathOptional.isEmpty()) {
+      printLine("No file was selected");
+      startButton.setDisable(true);
+      return;
+    }
+
+    String path = pathOptional.get();
+    printLine("Selected: " + path);
+    startButton.setDisable(false);
+
+    loadGame(path);
+    game.setOnEnd(() -> show(stage));
```

Comment:
It look like when `loadGame` fails (`game` field is not initialized) then we crash by `NullPointerException` in line 108. Maybe make `loadGame` return the `Result` and react to it here?
```suggestion
    loadGame(path).ifOkOrElse(ok -> { game = ok; game.setOnEnd... }, err -> { /* noop  or handle the error ? */}
```

Reply:
:thumbsup: